pkg/collector/corechecks/servicediscovery/module/ignore_proc.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package module
+
+import (
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// ignoreServices is a list of service names that should not be reported as a service.
+var ignoreServices = map[string]struct{}{
+	"datadog-agent": {},
+}
+
+// shouldIgnorePid returns true if service should be excluded from handling.
+func (s *discovery) shouldIgnorePid(proc *process.Process) bool {
+	s.mux.Lock()
+	_, found := s.ignorePids[proc.Pid]
+	s.mux.Unlock()
+
+	return found
+}
```

Comment:
If the only thing you need from the process is the pid, and the function is called shouldIgnore**Pid**, the input argument should be pid

Reply:
I've changed code